src/core/annotation.js

```diff
@@ -139,6 +141,70 @@ var Annotation = (function AnnotationClosure() {
 
   Annotation.prototype = {
     /**
+     * Set the flags.
+     *
+     * @public
+     * @memberof Annotation
+     * @param {number} flags - unsigned 32-bit integer specifying annotation
+     *                         characteristics
+     * @see {@link shared/util.js}
+     */
+    setFlags: function Annotation_setFlags(flags) {
+      if (flags === (flags | 0)) {
+        this.flags = flags;
+      } else {
+        this.flags = 0;
+      }
+    },
+
+    /**
+     * Check if a provided flag is set.
+     *
+     * @public
+     * @memberof Annotation
+     * @param {number} bitMask - hexadecimal representation for an annotation
+     *                           characteristic
+     * @return {boolean}
+     * @see {@link shared/util.js}
+     */
+    hasFlag: function Annotation_hasFlag(bitMask) {
+      if (this.flags) {
+        return (this.flags & bitMask) > 0;
+      }
+      return false;
+    },
+
+    /**
+     * Check if the annotation is viewable.
+     *
+     * @public
+     * @memberof Annotation
+     * @return {boolean}
+     */
+    isViewable: function Annotation_isViewable() {
+      if (this.flags) {
+        return !this.hasFlag(AnnotationFlag.INVISIBLE) &&
+               !this.hasFlag(AnnotationFlag.HIDDEN) &&
+               !this.hasFlag(AnnotationFlag.NOVIEW);
+      }
+      return true;
+    },
+
+    /**
+     * Check if the annotation is printable.
+     *
+     * @public
+     * @memberof Annotation
+     * @return {boolean}
+     */
+    isPrintable: function Annotation_isPrintable() {
```

Comment:
Same question as in https://github.com/mozilla/pdf.js/pull/6672/files#r45551969.